### PR TITLE
Feature/plugin generator

### PR DESCRIPTION
**Pull request by jakeberesford-palmetto**
## Pull request checklist

Before submitting your PR, please review the following checklist:

- [ ] Consider adding a unit test if your PR resolves an issue.
- [x] All new and existing tests pass locally (`palm test`)
- [x] Lint (`palm lint`) has passed locally and any fixes were made for failures
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Breaking changes

- [ ] Check if this pull request introduces a breaking change

## What does this implement/fix? Explain your changes.

Make it easier to create new palm plugins!

### CODE REVIEW

**Comment by jakeberesford-palmetto** [comment]
I should update docs before this PR is merged!

File: palm/plugins/core/templates/plugin/setup.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+this_directory = Path(__file__).parent
+long_description = Path(this_directory, 'README.md').read_text()
+
+setup(
+    name='palm-{{plugin_name}}',
+    version='0.0.1',
+    description='{{plugin_name}} extension for Palm CLI',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    author='Palmetto - Data & Analytics team',
```
Comment (ethanknox-palmetto) [inline palm/plugins/core/templates/plugin/setup.py:28]: nitt: do we want to prompt for this maybe?

Comment (ethanknox-palmetto) [reply palm/plugins/core/templates/plugin/setup.py:28]: or like "your-cool-team@something.com"

**Review by ethanknox-palmetto — APPROVED** [review]
nice additional feature!

File: palm/plugins/core/templates/plugin/setup.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    author_email='data-analytics-team@palmetto.com',
+    url='',
+    packages=find_namespace_packages(include=['palm', 'palm.*']),
+    package_data={'': ['*.yaml', '*.yml']},
```
[the remark this answers is not in the document]
Comment (jakeberesford-palmetto) [reply palm/plugins/core/templates/plugin/setup.py:32]: I was on the fence about putting anything in here. Really, that is an implementation detail. It's reasonable to think that a plugin may not need any package_data... I left this here mostly as a hint for anyone who does want to use it.

File: palm/plugins/core/templates/plugin/setup.py (hunk 1 above)
Comment (jakeberesford-palmetto) [reply palm/plugins/core/templates/plugin/setup.py:28]: Yeah, good shout, I'll bake this in to the generator

File: palm/plugins/core/commands/cmd_plugin.py [hunk 3]
```diff
@@ -7,6 +8,52 @@ def cli():
...
+@click.option(
+    "--author-email", multiple=False, help="Email address of the plugin author"
+)
+@click.pass_context
+def new(ctx, name: str, author: Optional[str], author_email: Optional[str]):
+    """
+    Generate a new plugin
+    """
+    if not author:
+        author = click.prompt("What is the author name")
```
Comment (mrogers-palmetto) [inline palm/plugins/core/commands/cmd_plugin.py:23]:
```suggestion
        author = click.prompt("What is the author name?")
```

**Review by mrogers-palmetto — APPROVED** [review]
Everything looks great. Super useful!

**Pull request merged**